graviti/dataset/dataset.py

```diff
@@ -103,14 +110,32 @@ def factory_getter(
                 128,
                 factory_getter,
             )
-            # catalog = get_catalog(
-            # self._url,
-            # self._access_key,
-            # self._dataset_id,
-            # draft_number=self._status.draft_number,
-            # commit=self._status.commit_id,
-            # )
-            self._data[sheet_name] = DataFrame.from_lazy_factory(factory)
+            catalog = get_catalog(
+                self._url,
+                self._access_key,
+                self._dataset_id,
+                commit=self._commit_id,
+            )
+            first_data_details = data_details["dataDetails"][0]
+            remote_data = RemoteData.from_response_body(
+                first_data_details,
+                url=URL(first_data_details["url"], lambda: "update is not supported currently"),
```

Comment:
The same code exists in `schema/extravotrs.py`, can we reuse that?

Reply:
do you mean to reuse `_URL` in `schema/extractors.py`?

Reply:
Yes.